armsrc/appmain.c

```diff
@@ -1158,9 +1158,14 @@ void UsbPacketReceived(uint8_t *packet, int len)
 		case CMD_READER_ISO_15693:
 			ReaderIso15693(c->arg[0]);
 			break;
+
 		case CMD_SIMTAG_ISO_15693:
 			SimTagIso15693(c->arg[0], c->d.asBytes);
 			break;
+
```

Comment:
Is there a specific reason for adding CMD_CSETUID_ISO_15693 on the device? Couldn't we use 4 calls with CMD_READER_ISO_15693 instead?

Reply:
Hi piwi,

I'm not sure to understand...
How could I use CMD_READER_ISO_15693 since I have to write on the card, not to read it.
Maybe I'm missing something.

Did you mean CMD_ISO_15693_COMMAND ?
The 4 commands have to be sent in a row but since the `DirectTag15693Command()` function (used by CMD_ISO_15693_COMMAND command) has a protection against rdv4.0 overheating (`FpgaWriteConfWord(FPGA_MAJOR_MODE_OFF);`), it can't be used that way (4 calls to CMD_ISO_15693_COMMAND one after the other).
That's why the lua script doesn't work on the official repo anymore, by the way...

I was about to duplicate the `DirectTag15693Command()` without this protection so I could use it with 4 calls but I thought it would be counter-productive to give everyone an access to a rdv4.0-unsafe command, that's why I created a specific command for that.
And that would have been a new command creation too, anyhow.

Best regards.

Reply:
You are correct. Sorry, I shouldn't do code reviews on smartphone :smile:.